bootstrap.py

```diff
@@ -268,8 +278,19 @@ def post_json(url, jdata):
         request.add_header("Accept", "application/json")
         request.get_method = lambda: 'POST'
         reply = opener.open(request)
+    except urllib2.HTTPError, e:
+        if e.code == 401:
```

Comment:
I would drop all these hooks and simply raised one general HTTPError with printing out the `e.read` as the api itself contains the error message, e.g.:
```
error: {
  "error": {"message":"Unable to authenticate user admin"}
}
```
- I think it is self explaining and does not require you to handle all sorts of return codes.